internal/api/handlers/url.go

```diff
@@ -0,0 +1,71 @@
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/GlebRadaev/shlink/internal/config"
+	"github.com/GlebRadaev/shlink/internal/service"
+	"github.com/go-chi/chi/v5"
+)
+
+// URLHandlers defines the handlers for URL shortening
+type URLHandlers struct {
+	urlService *service.URLService
+	config     *config.Config
+}
+
+// NewURLHandlers creates a new instance of URLHandlers
+func NewURLHandlers(urlService *service.URLService, config *config.Config) *URLHandlers {
+	return &URLHandlers{
+		urlService: urlService,
+		config:     config,
+	}
+}
+
+// Shorten handles the request to shorten a URL
+func (h *URLHandlers) Shorten(w http.ResponseWriter, r *http.Request) {
+	contentType := r.Header.Get("Content-Type")
+	parts := strings.Split(contentType, ";")
+	if parts[0] != "text/plain" {
+		http.Error(w, "Invalid content type", http.StatusBadRequest)
+		return
+	}
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Failed to read request body", http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+	url := string(body)
+	shortURL, err := h.urlService.Shorten(url)
+	log.Print(err)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	shortID := fmt.Sprintf("%s/%s", h.config.BaseURL, shortURL)
```

Comment:
[CHANGE REQUEST] Split the application into layers
https://github.com/golang-standards/project-layout
https://medium.com/@deanrubin/the-three-layered-architecture-fe30cb0e4a6#:~:text=The%20presentation%20layer%20is%20the,Update%20or%20Delete)%20operation%20flow.